table/merging_iterator.cc

```diff
@@ -188,6 +212,20 @@ class MergingIterator : public InternalIterator {
       minHeap_.pop();
     }
     current_ = CurrentForward();
+
+    if (range_del_agg_ != nullptr && Valid()) {
+      // When we cross over to a new range tombstone, we use an optimization to
+      // seek the iterators whose keys are all covered by the range tombstone.
+      // TODO(ajkr): this must be optimized to seek only when tombstone changed.
+      auto optimized_seek_endpoint = range_del_agg_->GetEndpoint(
+          key(), RangeDelPositioningMode::kForwardTraversal);
+      if (optimized_seek_endpoint.endpoint() != nullptr &&
+          optimized_seek_endpoint.seq() > 0) {
+        std::string endpoint_ikey;
+        AppendInternalKey(&endpoint_ikey, *optimized_seek_endpoint.endpoint());
+        SeekIfSeqnoSmaller(endpoint_ikey, optimized_seek_endpoint.seq());
```

Comment:
Just thinking out loud here: this might result in a key that's covered by another range tombstone, not sure if we want to make it a loop here (on the other hand, too many seek slows `Next()` down).